beamer/agent/events.py

```diff
@@ -307,10 +314,11 @@ def fetch(self) -> list[Event]:
 
         result = []
         from_block = self._next_block_number
+
         while from_block <= block_number:
             to_block = min(block_number, BlockNumber(from_block + self._blocks_to_fetch))
             try:
-                events = self._fetch_range(from_block, to_block)
+                events = self._fetch_range(from_block, BlockNumber(to_block))
```

Comment:
Why do we need to cast `to_block` to `BlockNumber`? It was without it before...

Reply:
my bad :D forgot it